18F/identity-idp
# LG-7185: Send email reminders for in-person proofing enrollments

[LG-7185](https://cm-jira.usa.gov/browse/LG-7185)

🛠 Summary of changes
This pr is to add the job for the email reminder. This job is to run daily. The job checks enrollments that fit the database query in needs_email_reminders - these are enrollments that have a pending status, have an established_at date and have at least 11 days left till their enrollment period ends.

📜 Testing Plan
Provide a checklist of steps to confirm the changes.

 Open the rails console
 1. Save your latest enrollment, `e = InPersonEnrollment.last`
 2. Update the enrollment so it has 11 days left:
```ruby
# assuming enrollment_established_at == created_at
e.update!(enrollment_established_at: e.created_at - 19.days)
# check how many days are left
e.days_to_due_date
=> 11
```
 3. Update enrollment status to "pending" if it isn't already, `e.update(status: "pending")`
 4. Trigger job:  `InPerson::EmailReminderJob.new.perform(Time.zone.now)`
 5. Check that you received an email in mailcatcher
 6. Update enrollment so it now has 4 days left.
```ruby
# assuming enrollment_established_at == created_at
e.update!(enrollment_established_at: e.created_at - 26.days)
# check how many days are left
e.days_to_due_date
=> 4
```
 8. Trigger job again and confirm email was sent to mailcatcher

## 👀 Screenshots

<details>
<summary>en- 10 days:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/204347225-7b6175f0-86fa-4d57-a31f-8ddfe747974d.png">
</details>

<details>
<summary>en- 3 days:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/203643580-0f1b2537-1f8a-4a8b-8c06-df48b2701542.png">
</details>

<details>
<summary>en- 1 day:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/203643599-d4ca74ca-378d-4ca0-96d1-2fd0c2e778c4.png">
</details>

<details>
<summary>es- 10 days:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/204347340-cb17daba-43b7-4cb7-8559-00f6562ca2d0.png">
</details>

<details>
<summary>es- 3 days:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/203643653-bf6f45fd-03f0-4260-885c-8973df3ab268.png">
</details>

<details>
<summary>es- 1 day:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/203643677-22248884-2f3f-4ded-a19b-a13266ff3859.png">
</details>

<details>
<summary>fr- 10 days:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/204347408-db762a28-3f9f-4e7a-affc-d37313eb0c78.png">
</details>

<details>
<summary>fr- 3 days:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/203643719-03f5775a-4a27-4565-afaa-9990cdf7bb80.png">
</details>

<details>
<summary>fr- 1 day:</summary>
<img width="584" src="https://user-images.githubusercontent.com/45415133/203643733-7fec4fb4-74dd-40a4-826a-76591e6a3009.png">
</details>
